src/Navbar.js

```diff
@@ -1,91 +1,106 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import cx from 'classnames';
-import Col from './Col';
 import Icon from './Icon';
 
 class Navbar extends Component {
   constructor(props) {
     super(props);
-    this.renderSideNav = this.renderSideNav.bind(this);
   }
 
   componentDidMount() {
-    if (typeof $ !== 'undefined') {
-      $('.button-collapse').sideNav(this.props.options);
+    const { options } = this.props;
+
+    if (typeof M !== 'undefined') {
+      this.instance = M.Sidenav.init(this._sidenav, options);
     }
   }
 
-  renderSideNav() {
-    return (
-      <ul id="nav-mobile" className="side-nav">
-        {this.props.children}
-      </ul>
-    );
+  componentWillUnmount() {
+    if (this.instance) {
+      this.instance.destroy();
+    }
   }
 
   render() {
     const {
       brand,
       className,
+      extendsWith,
       fixed,
-      left,
-      right,
+      alignLinks,
       centerLogo,
       href,
+      children,
       ...other
     } = this.props;
 
     delete other.options;
 
-    let classes = {
-      right: right,
-      'hide-on-med-and-down': true
-    };
-
-    let brandClasses = {
+    let brandClasses = cx({
       'brand-logo': true,
-      right: left,
-      center: centerLogo
-    };
+      center: centerLogo,
+      left: alignLinks === 'right'
+    });
+
+    const navMobileCSS = cx('hide-on-med-and-down', [alignLinks]);
+
+    const links = React.Children.map((link, index) => (
+      <li key={index}>{link}</li>
+    ));
 
-    let content = (
+    let navbar = (
       <nav {...other} className={className}>
         <div className="nav-wrapper">
-          <Col s={12}>
-            <a href={href} className={cx(brandClasses)}>
+          {brand && (
+            <a href="/" className={brandClasses}>
               {brand}
             </a>
-            <ul className={cx(className, classes)}>{this.props.children}</ul>
-            {this.renderSideNav()}
-            <a className="button-collapse" href="#" data-activates="nav-mobile">
-              <Icon>view_headline</Icon>
-            </a>
-          </Col>
+          )}
+          <a href="#!" data-target="mobile-nav" className="sidenav-trigger">
+            <Icon>menu</Icon>
+          </a>
+          <ul className={navMobileCSS}>{links}</ul>
         </div>
+        {extendsWith && (
```

Comment:
as per the [docs](https://materializecss.com/navbar.html) if we extend the navbar, it should be wrapped in an additional classname